segmentstore/server/src/main/java/io/pravega/segmentstore/server/containers/ContainerRecoveryUtils.java

```diff
@@ -0,0 +1,210 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.segmentstore.server.containers;
+
+import com.google.common.base.Preconditions;
+import io.pravega.common.Exceptions;
+import io.pravega.common.concurrent.Futures;
+import io.pravega.segmentstore.contracts.SegmentProperties;
+import io.pravega.segmentstore.contracts.StreamSegmentNotExistsException;
+import io.pravega.segmentstore.contracts.tables.IteratorArgs;
+import io.pravega.segmentstore.server.tables.ContainerTableExtension;
+import io.pravega.segmentstore.storage.Storage;
+import io.pravega.shared.NameUtils;
+import io.pravega.shared.segment.SegmentToContainerMapper;
+import lombok.extern.slf4j.Slf4j;
+import lombok.val;
+
+import java.io.IOException;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.stream.Collectors;
+
+import static io.pravega.shared.NameUtils.getMetadataSegmentName;
+
+/**
+ * Utility methods for container recovery.
+ */
+@Slf4j
+public class ContainerRecoveryUtils {
+    private static final Duration TIMEOUT = Duration.ofSeconds(30);
+
+    /**
+     * This method lists the segments from the given storage instance. It then registers all segments except Attribute
+     * segments to the container metadata segment(s).
+     * {@link DebugStreamSegmentContainer} instance(s) are provided to this method which can have some segments already present
+     * in their respective container metadata segment(s). After the method successfully completes, only the segments which
+     * existed in the {@link Storage} will remain in the container metadata. All segments which only existed in the container
+     * metadata or which existed in both container metadata and the storage but with different lengths and/or sealed status,
+     * will be deleted from the container metadata. If the method fails while execution, appropriate exception is thrown.
+     * All segments from the storage are listed one by one, then mapped to their corresponding {@link DebugStreamSegmentContainer}
+     * instances for registering them to container metadata segment.
+     * @param storage                           A {@link Storage} instance that will be used to list segments from.
+     * @param debugStreamSegmentContainers      A Map of Container Ids to {@link DebugStreamSegmentContainer} instances
+     *                                          representing the containers that will be recovered.
+     * @param executorService                   A thread pool for execution.
+     * @throws InterruptedException             Required for Futures.get()
```

Comment:
For all these methods, do this for the exceptions:
- Have them declare that they throw `Exception`.
- Do not document `InterruptedException` or `ExecutionException`.
- For `TimeoutException` and `IOException` please have appropriate descriptions. `Required for Futures.get` is not of any use for the user.

Reply:
Done.